src/msg.c

```diff
@@ -1065,7 +754,1282 @@ void draw_message_window(MessagePrintState* printer) {
+    s16 phi_t5;
+    s16 phi_s0_6;
+    s16 phi_t3;
+    s32 dbPos;
+    s16 charPosX;
+    s16 charPosY;
+    s16 phi_s2_5;
+    s32 phi_s0_7;
+    s32 phi_s1_8;
+    s8 varImgHasBorder;
+    s16 t1;
+    s16 t2;
+    f32 z1;
+    f32 z2;
+    f32 temp1;
+    f32 temp2;
+
+    msg_drawState = &D_80155D20;
+    msg_drawState->printBuffer = printer->printBuffer;
+    msg_drawState->printModeFlags = 0;
+    if (gMasterGfxPos != D_80151338) {
+        msg_reset_gfx_state();
+    }
+    msg_drawState->printModeFlags |= 0x100 | 1;
+    sp8E = 0xFF;
+    if (flag & 1) {
+        sp8E = alpha;
+    }
+    msg_drawState->textStartPos[0] = 0;
+    msg_drawState->textStartPos[1] = 0;
+    msg_drawState->clipX[0] = 0;
+    msg_drawState->clipY[0] = 0;
+    msg_drawState->clipX[1] = SCREEN_WIDTH - 1;
+    msg_drawState->clipY[1] = SCREEN_HEIGHT - 1;
+    msg_drawState->msgScale.x = 1.0f;
+    msg_drawState->msgScale.y = 1.0f;
+    msg_drawState->unk_29 = 0;
+    msg_drawState->framePalette = 0;
+    msg_drawState->unk_2C = 0;
+    msg_drawState->visiblePrintedCount = 0;
+    msg_drawState->centerPos = 0;
+    msg_drawState->fontVariant = 0;
+    msg_drawState->currentPosX = 0;
+    msg_drawState->nextPos[0] = 0;
+    msg_drawState->nextPos[1] = 0;
+    msg_drawState->font = 0;
+    msg_drawState->effectFlags = 0;
+    msg_drawState->nextCounter = 0;
+    msg_drawState->textColor = 0;
+    sp96 = 0xFF;
+    spAE = 0xFF;
+    spB6 = 0;
+    spB8 = 0;
+    sp9E = -1;
+
+    if (flag & 2) {
+        msg_drawState->font = 1;
+    }
+
+    if (flag & 4) {
+        gDPSetScissor(gMasterGfxPos++, G_SC_NON_INTERLACE, 0, 0, SCREEN_WIDTH - 1, SCREEN_HEIGHT - 1);
+    }
+
+    sp80bool = FALSE;
+    msg_drawState->drawBufferPos = 0;
+
+    do {
+        switch (msg_drawState->printBuffer[msg_drawState->drawBufferPos]) {
+            case 0xF0:
+                msg_drawState->nextPos[0] = 0;
+                msg_drawState->nextPos[1] += (s32) ((msg_drawState->msgScale.y * gMsgCharsets[msg_drawState->font]->newLineY) + D_802EB644[printer->style]);
+                if (msg_drawState->printModeFlags & 0x40) {
+                    msg_drawState->printModeFlags |= 0x80;
+                }
+                msg_drawState->drawBufferPos++;
+                break;
+            case 0xF9:
+                msg_drawState->drawBufferPos++;
+                break;
+            case 0xF1:
+            case 0xF2:
+            case 0xF3:
+            case 0xF4:
+                msg_drawState->fontVariant = msg_drawState->printBuffer[msg_drawState->drawBufferPos] - 0xF1;
+                msg_drawState->drawBufferPos++;
+                break;
+            case 0xF5:
+            case 0xF6:
+            case 0xF7:
+                msg_drawState->nextPos[0] += msg_get_draw_char_width(msg_drawState->printBuffer[msg_drawState->drawBufferPos], msg_drawState->font, msg_drawState->fontVariant, msg_drawState->msgScale.x, msg_drawState->currentPosX, msg_drawState->printModeFlags);
```

Comment:
Some lines in this PR seem overly long. This one totals in at 250 characters before even looking at the indentation!